lib/ex_csv/parser.ex

```diff
@@ -82,10 +82,14 @@ defmodule ExCsv.Parser do
   end
 
   # NEWLINE
+  defp build(<<rt,nl>> <> rest, [[current_field | previous_fields] | previous_rows], %{return: rt, newline: nl, quoting: false} = config) do
+    build_newline(rest, current_field, previous_fields, previous_rows, config)
+  end
+  defp build(<<char>> <> rest, [[current_field | previous_fields] | previous_rows], %{return: char, quoting: false} = config) do
```

Comment:
The guard version _should_ match lines 88 and 91, which match the return character and newline character respectively.  Line 85 has to stay.